src/gui/invalidfilenamedialog.cpp

```diff
@@ -0,0 +1,168 @@
+#include "invalidfilenamedialog.h"
+#include "accountfwd.h"
+#include "common/syncjournalfilerecord.h"
+#include "propagateremotemove.h"
+#include "ui_invalidfilenamedialog.h"
+
+#include <folder.h>
+
+#include <QPushButton>
+#include <QDir>
+#include <qabstractbutton.h>
+#include <QDialogButtonBox>
+#include <QFileInfo>
+#include <QPushButton>
+
+namespace {
+const QVector<QChar> illegalCharacters({ '\\', '/', ':', '?', '*', '\"', '<', '>', '|' });
```

Comment:
replace by `constexpr std::array<QChar, 9>` to avoid any kind of memory management problems and remove one dynamic memory allocation at runtime